daemon/api/dependencies.py

```diff
@@ -0,0 +1,185 @@
+from pathlib import Path
+from typing import Dict, List, Optional
+from pydantic import FilePath
+from pydantic.errors import PathNotAFileError
+from fastapi import HTTPException, UploadFile, File
+
+from jina import __default_host__, Flow
+from jina.enums import PeaRoleType, SocketType
+from jina.helper import ArgNamespace, cached_property, random_port
+
+from .. import __dockerhost__
+from ..helper import get_workspace_path
+from ..models.enums import WorkspaceState
+from ..stores import workspace_store as store
+from ..models import DaemonID, FlowModel, PodModel, PeaModel
+
+
+class FlowDepends:
+    def __init__(self, workspace_id: DaemonID, filename: str) -> None:
+        self.workspace_id = workspace_id
+        self.filename = filename
+        self.localpath = self.validate()
+        self.id = DaemonID('jflow')
+        self.params = FlowModel(
+            uses=self.filename, workspace_id=self.workspace_id.jid, identity=self.id
+        )
+
+    @cached_property
```

Comment:
why does it need to be a cached property? can't it be simply a `property`? or can we just cache it ourselves? I think the extra work done by `cached_property` is unwanted

Reply:
I've been using `cached_property` as a setter & a getter. We'll be moving away from `cached_property` wherever not required in the next set of cleanup PRs.